### ISSUE TITLE
[Assert] Moving elements in PostFX Layer component results in duplicating an element

### ISSUE THREAD

**Issue by LB-JakubSkorupka** [o3de/o3de#7516]
**Describe the bug**
When moving elements in the **PostFX Layer** component, one of the elements is duplicated in place of another element.
Drag and dropping the element with an index greater than [0] on the element with index [0] will result in duplicating that ([0]) element and replacing the next (in ascending order) element that is **not** already a duplicate of [0] element.
Similar behavior can be observed for dragging and dropping an element on the element with the highest (last) index.

On the **Linux** OS, two asserts are printed in the console when reproducing this issue.
Please note that the asserts are **not** printed in the console on **Windows** OS.

**Assert 1**

**File**:
`/home/gpc-XXX/temp/o3de/Code/Framework/AzCore/./AzCore/Serialization/AZStdContainers.inl`

**Condition**:

```
(140307444803840) 'virtual void *AZ::Internal::NullFactory::Create(const char *)'
(System) - Containers and pointers (AZStd::string) should be stored by value. They are made to be small and have minimal overhead when empty!
```

**Line**: `115`

**Assert 2**

**File**:
` /home/gpc-XXX/temp/o3de/Code/Framework/AzCore/AzCore/Serialization/SerializeContext.cpp`

**Condition**:
```
(140307444803840) 'void AZ::SerializeContext::CloneObjectInplace(void *, const void *, const AZ::Uuid &)'
(System) - SerializeContext::CloneObjectInplace - Attempt to clone a nullptr.
```

**Line**: `2100`

**Steps to reproduce**
1. Create an entity.
2. Add a **PostFX Layer** component to the created entity.
3. Add four "Select Camera Tags Only" elements in the **PostFX Layer** component.
4. Name the elements as follows: 1/2/3/4 (in order they were created, for better visualization of the issue).
5. Grab [0] element and drop it on the [3] element to change that element's position.

**Expected behavior**
Moving the elements in **PostFX Layer** component is consistent, no duplicate elements are created, **no** asserts are printed in the console.

**Actual behavior**
Moving the elements in **PostFX Layer** component results in duplicating an element, asserts are printed in the console.

**Video**

https://user-images.githubusercontent.com/86951967/153216617-525cc9c9-0048-4aca-9798-71d82fb70d7e.mp4

**Found in Branch**
Development (3f32669)

**Desktop:**
 - Device: PC
 - OS: Linux
 - Version: Ubuntu 20.04.3 LTS
 - CPU: Intel Core i7-8700K
 - GPU: NVIDIA GeForce RTX 2080 Ti
 - Memory: 16GB

**Additional context**
[Editor.log](https://github.com/o3de/o3de/files/8032919/Editor.log)

**Comment by LB-JakubSkorupka** [comment]
On build Development (6df71ce), asserts printed in the console differ from the ones mentioned in the issue when reproducing the issue:
 - The **Line** in **Assert 1** changed from 115 to 113.
 - In both asserts, in **Condition**, The number present in parentheses changed from 140307444803840 to 14036.
Please note that the issue also occurs when using **Tag** component instead of PostFX Layer component.

**Comment by LB-JakubSkorupka** [comment]
Issue verified as fixed on build Development (61f1dc1).
